core/src/main/java/org/testcontainers/containers/Network.java

```diff
@@ -52,17 +53,17 @@ static Network newNetwork() {
         @Singular
         private Set<Consumer<CreateNetworkCmd>> createNetworkCmdModifiers;
 
-        private String id;
+        private final AtomicReference<String> id = new AtomicReference<>();
```

Comment:
1) both `getId` and `close` are `synchronized`, the field can be `volatile`
2) by making it final, you're breaking the backward compatibility, since (accidentally) the constructor was exposed to public (a class that was intended to be internal became private as it was defined in the interface), please revert

Reply:
I see, I thought that `NetworkImpl` is internal, but it indeed leaks out through the builder

Reply:
FYI:
soon we will add japicmp to automate such checks

Reply:
also, please do not unresolve other's comments.